homeassistant/helpers/intent.py

```diff
@@ -253,19 +269,40 @@ def __init__(self, intent: Intent | None = None) -> None:
         self.reprompt: dict[str, dict[str, Any]] = {}
         self.card: dict[str, dict[str, str]] = {}
 
+    @property
+    def language(self) -> str | None:
+        """Return intent language if available."""
+        return self.intent.language if self.intent is not None else None
+
     @callback
     def async_set_speech(
-        self, speech: str, speech_type: str = "plain", extra_data: Any | None = None
+        self,
+        speech: str,
+        speech_type: str = "plain",
+        extra_data: Any | None = None,
+        language: str | None = None,
```

Comment:
I would expect language to be set at response level. Maybe even just from intent? Why allow passing it in

Reply:
I'll just set it from the intent only :+1: